core/engine.go

```diff
@@ -112,23 +102,19 @@ func (e *Engine) Init(globalCtx, runCtx context.Context) (run func() error, wait
 	}
 
 	// TODO: move all of this in a separate struct? see main TODO above
-	runSubCtx, runSubCancel := context.WithCancel(runCtx)
-
-	execRunResult := make(chan error)
-	engineRunResult := make(chan error)
 	processMetricsAfterRun := make(chan struct{})
 	runFn := func() error {
 		e.logger.Debug("Execution scheduler starting...")
-		err := e.ExecutionScheduler.Run(globalCtx, runSubCtx, e.Samples)
-		e.logger.WithError(err).Debug("Execution scheduler terminated")
-
-		select {
-		case <-runSubCtx.Done():
-			// do nothing, the test run was aborted somehow
-		default:
-			execRunResult <- err // we finished normally, so send the result
+		err := e.ExecutionScheduler.Run(globalCtx, runCtx, e.Samples)
+		if err == nil {
+			e.logger.Debug("Execution scheduler finished nominally")
```

Comment:
It doesn't really matter, since this will be removed in #2815, but I think you mean "normally" here. "Nominally" means [something else](https://dictionary.cambridge.org/dictionary/english/nominally).

Reply:
Resolved by [this force-push](https://github.com/grafana/k6/compare/cb178421219554030f257949e7ca4259f467ba38..c23495567d2685df979385eabf7cb050df9440cd).